lib/collection/src/common/stoppable_task.rs

```diff
@@ -28,9 +32,48 @@ impl<T> StoppableTaskHandle<T> {
         self.ask_to_stop();
         self.is_started().then_some(self.join_handle)
     }
+
+    /// Join this stoppable task and handle any panics
+    ///
+    /// Any panics are propagated through the configured panic handler. If no handler is
+    /// configured, nothing happens.
+    ///
+    /// To call this, the task must already be finished. Otherwise it panics in development, or
+    /// blocks in release.
```

Comment:
Could have have a large timeout instead of blocking forever?

Reply:
Yeah. But this method should not be called before the task is finished anyway. That's why there's a (debug) assert here to ensure that.